.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: double-quote-string-fixer
+      - id: detect-private-key
+
+  - repo: https://github.com/google/yapf
+    rev: v0.40.2
+    hooks:
+      - id: yapf
```

Comment:
in other projects we use black, can you switch to it ?
https://black.readthedocs.io/en/stable/integrations/source_version_control.html

Reply:
Should we really switch to black? Both yaptide and pymchelper use yapf. Also, in my opinion, the line length limitation in black often makes the code less readable.

Reply:
Example:

YAPF:
```python
if "worldZone" in zones_json and "averageIonisationPotential" in zones_json["worldZone"].get(
            "materialPropertiesOverrides", []):
        ionisation_potentials.append(
            IonisationPotential(
                material_name=materials[zones_json["worldZone"]["materialUuid"]].fluka_name,
                ionisation_potential=zones_json["worldZone"]["materialPropertiesOverrides"]
                ["averageIonisationPotential"],
            ))
```									

BLACK:

```python
if "worldZone" in zones_json and "averageIonisationPotential" in zones_json[
        "worldZone"
    ].get("materialPropertiesOverrides", []):
        ionisation_potentials.append(
            IonisationPotential(
                material_name=materials[
                    zones_json["worldZone"]["materialUuid"]
                ].fluka_name,
                ionisation_potential=zones_json["worldZone"][
                    "materialPropertiesOverrides"
                ]["averageIonisationPotential"],
            )
        )
```

Reply:
You are correct, yapf is used there. But don't forget to configure line length in yapf. 
See https://github.com/yaptide/yaptide/blob/0d0f76a178268cae75bfe966d37b1561503e6dea/setup.cfg#L7

Both yapf and black use ~80 characters by default. And both can be configured to accept longer lines